src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -51,6 +51,18 @@
 
     /**
      * Constructs an unauthenticated BoxAPIRequest.
+     *
+     * @param url
+     *            the {@link URL} of the request
+     * @param method
+     *            the {@link HttpMethod} of the request
+     */
+    public BoxAPIRequest(URL url, HttpMethod method) {
```

Comment:
Yeah, we can always come back to this later. I'd prefer if it was left as a String for now.